### PR TITLE
Sync csproj when files are changed from the Godot FileSystem dock

### LINKED ISSUE #12917: Renaming a .cs file does not edit the .csproj file
**Operating system or device, Godot version, GPU Model and driver (if graphics related):**
Windows 8.1 - custom Godot build (2cdfef5)

**Issue description:**
When you rename a .cs file from within the editor, the .csproj file doesn't get changed. This makes it so you have to open the mono project file manually to change the filename there too before you can run your Godot project again.

**Steps to reproduce:**
1. Create a new project
2. Add a C# file
3. Change that file's name to something else

### PR DESCRIPTION
**Pull request by neikeq**
Closes #12917

I had to expose `FileSystemDock` to the scripting API, which partially addresses #26709 (although that issue should have been two separate issues).

### CODE REVIEW

**Comment by neikeq** [comment]
@akien-mga This cannot be cherry-picked directly due to the new way of using signals in C#. I can make a separate PR against the 3.2 branch. WDYT?

**Comment by neikeq** [comment]
Related to #37116. :eyes:

**Comment by neikeq** [comment]
@akien-mga Alternatively, I can separate this into two commits. Most of the code in the first one, compatible with 3.2, and the second one only updating to the new signal events syntax. This way you can simply cherry-pick the first commit.

File: editor/filesystem_dock.cpp [hunk 1]
```diff
@@ -1288,11 +1288,11 @@ void FileSystemDock::_make_scene_confirm() {
 }
 
 void FileSystemDock::_file_deleted(String p_file) {
-	emit_signal("file_deleted", p_file);
+	emit_signal("file_removed", p_file);
 }
 
 void FileSystemDock::_folder_deleted(String p_folder) {
-	emit_signal("folder_deleted", p_folder);
+	emit_signal("folder_removed", p_folder);
```
Comment (akien-mga) [inline editor/filesystem_dock.cpp]: These might have been better in the first commit that exposes FileSystemDock and could be cherry-picked easily, but I guess it's fine as is if you plan to make a matching 3.2 PR. No current code relies on those signals and the class wasn't exposed yet, so the bug couldn't be triggered.

Comment (akien-mga) [reply editor/filesystem_dock.cpp]: Might be worth renaming the methods too for consistency.

**Comment by akien-mga** [comment]
> This cannot be cherry-picked directly due to the new way of using signals in C#. I can make a separate PR against the 3.2 branch. WDYT?

A dedicated PR for 3.2 sounds good.

File: editor/filesystem_dock.cpp (hunk 1 above)
Comment (neikeq) [reply editor/filesystem_dock.cpp]: As suggested, moved these changes to the first commit and renamed the methods as well.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**